### ISSUE TITLE
Feature Request: In-Mod method of creating Grass Path

### ISSUE THREAD

**Issue by Birgliss** [mickelus/tetra#56]
## Feature Request

**Feature description**

I think there should be a way to create grass path. In vanilla, you do this with a shovel, but the Tetra equivalent (the Adze) has no way to do this.
However, I tend to use the Adze in conjunction with the Hoe. Adding a right-click function to it would be risky, and could be a pain to implement. As well, the Adze already has 2 functions, adding another could potentially be overkill. 
I recommend adding it to the Hammer, as it still makes thematic sense, and gives the hammer a use outside of crafting and scavenging material. Additionally, in my experience, I tend to use the hammer in both-sides of the tool, reducing chance of edge-cases with the hoe. 
Though, if this does arise, I recommend it applies grass path first, and then can hoe the path into tilled soil. (This is a function that already exists in Vanilla.)

**How it improves the player experience**

I think it's important for tools from Tetra to have complete parity with Vanilla tool functionality, so a player could choose to switch over to them entirely. As well, Grass Path appears to be stamped-down dead grass. It would make some amount of sense for a hammer to do this. 

**Tetra synergies**

As said above, it would improve Tetra's feature parity with Vanilla. As far as I know, adding this would give complete feature parity in-comparison with Vanilla.

**Comment by mickelus** [comment]
I added functionality for this while I implemented tilling (for hoes) but I did not have a fitting module to add it to. If an item can both till and flatten it will flatten if the player is sneaking, otherwise till.

Perhaps it could be an improvement for the hammer, that way we could ease players into it rather than just dumping a bunch of new functionality on them when they craft the hammer (which is the first tetra item players will craft).

If you want to play around with having this on some modules, add the "flattening" effect to its configuration.
